docs/firmware/zephyr-device-sdk/light-db/guide-lightdb-get.md

```diff
@@ -2,95 +2,92 @@
 title: LightDB Get Request
 sidebar_position: 2
 ---
+
 The [LightDB Get sample
 application](https://github.com/golioth/golioth-zephyr-sdk/tree/main/samples/lightdb/get)
-demonstrates how to retrieve resources from LightDB State using the Golioth SDK.
-Data is retrieved asynchronously and sent to a handler function registered when
-the get request is first made.
-
-## Details
+demonstrates how to retrieve resources from LightDB State using the Golioth
+SDK. Data can be retrieved synchronously, or asyncronously. Synchronous
+requests block program flow until a response is received. Asynchronous requests
+register a callback function that runs when a response is received at some
+point in the future.
 
-### Preparation
+## Includes
 
 ```c
-#include <net/coap.h>
-static struct coap_reply coap_replies[1];
+#include <net/golioth/system_client.h>
 ```
 
-The Zephyr CoAP library provides a set of helper functions. Creating an array of
-`coap_reply` structs is necessary for registering the callback and pointing to
-the data when it arrives from the Golioth servers.
+Including the Golioth System Client header file makes the LightDB Get API
+functions available to your program.
 
-### Callback function
+## Synchronous Get
 
 ```c
-static int reply_callback(const struct coap_packet *response,
-                          struct coap_reply *reply,
-                          const struct sockaddr *from)
-{
-    /* Process the received message */
-}
-```
+uint8_t value[128];
+size_t len = sizeof(value);
+int err;
 
-A callback function is necessary to handle the reply received from the get
-request. Get requests can use different callbacks, and these functions may be
-arbitrarily named.
+err = golioth_lightdb_get(client, "counter",
+				  GOLIOTH_CONTENT_FORMAT_APP_JSON,
+				  value, &len);
+```
 
-### Route data to callback using `on_message`
+The simplest way to get LightDB State data from Golioth is using the
+synchronous `golioth_lightdb_get()` function. This blocking function will
+request data from the given endpoint ("counter" in this example).
 
-```c
-static void golioth_on_message(struct golioth_client *client,
-                               struct coap_packet *rx)
-{
-    coap_response_received(rx, NULL, coap_replies,
-                           ARRAY_SIZE(coap_replies));
-}
-```
+The returned value (or JSON object) is stored in a character array. A pointer
+to this array and the length of the array are passed as the final two
+parameters of the synchronous get function.
```

Comment:
It's probably worth pointing out that the `len` parameter is both an input and an output.

From Doxygen:

```
 * @param[in,out] len Size of buffer on input, size of response on output
```